### PR TITLE
Lysenko/ocd 1518 get rollout command

### PR DESCRIPTION
**Pull request by konstantin-lysenko-netapp**
# Jira Ticket

[OCD-1518](https://spotinst.atlassian.net/browse/OCD-1518)

### CODE REVIEW

File: pkg/oceancd/model/rollout/rollout.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+type VersionStatus struct {
+	Version           string       `json:"version"`
+	K8sService        string       `json:"k8sService"`
+	TrafficPercentage int          `json:"trafficPercentage"`
+	Replicas          ReplicasInfo `json:"replicas"`
+}
+
+type Rollout struct {
+	ID                        string        `json:"id"`
```
Comment (caduri) [inline pkg/oceancd/model/rollout/rollout.go:78]: change to `Id`

File: pkg/oceancd/model/rollout/rollout.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	NewRolloutId              string        `json:"newRolloutId"`
+	StartTime                 string        `json:"startTime"`
+	EndTime                   string        `json:"endTime"`
+	ClusterId                 string        `json:"clusterId"`
+	Namespace                 string        `json:"namespace"`
+	Strategy                  string        `json:"strategy"`
+	HasBackgroundVerification bool          `json:"hasBackgroundVerification"`
+	NewVersionStatus          VersionStatus `json:"newVersionStatus"`
+	StableVersionStatus       VersionStatus `json:"stableVersionStatus"`
+	Phases                    []Phase       `json:"phases"`
```
Comment (caduri) [inline pkg/oceancd/model/rollout/rollout.go:91]: This is not part of Rollout object
I think you should create a wrapper called DetailedRollout that includes Rollout and Phase list
wdyt?

File: pkg/utils/signals.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package utils
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+func SetupSignalHandler(cancel context.CancelFunc) {
```
Comment (caduri) [inline pkg/utils/signals.go:10]: Nice!
did you copy it from somewhere?

File: viewcontroller/converter/converter.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
+package converter
```
Comment (caduri) [inline viewcontroller/converter/converter.go:1]: if you copied any of the methods here, please add a comment with credit

File: viewcontroller/viewcontroller.go [hunk 5]
```diff
@@ -0,0 +1,430 @@
+package viewcontroller
```
Comment (caduri) [inline viewcontroller/viewcontroller.go:1]: Great work!

If you copied something here please add comment with credit

**Review by caduri — CHANGES_REQUESTED** [review]

File: pkg/oceancd/model/rollout/rollout.go (hunk 2 above)
Comment (konstantin-lysenko-netapp) [reply pkg/oceancd/model/rollout/rollout.go]: I think so

File: pkg/utils/signals.go (hunk 3 above)
Comment (konstantin-lysenko-netapp) [reply pkg/utils/signals.go:12]: added

File: viewcontroller/converter/converter.go (hunk 4 above)
Comment (konstantin-lysenko-netapp) [reply viewcontroller/converter/converter.go:1]: no

File: viewcontroller/viewcontroller.go (hunk 5 above)
Comment (konstantin-lysenko-netapp) [reply viewcontroller/viewcontroller.go:1]: added

**Review by caduri — APPROVED** [review]

**Pull request merged**